NewForce-Cohort-5/tabloidfullstack-fool-s-spring
# Ajb tag edit

# Description

This is a new feature that allows user edit an existing tag in the database that is displayed through the browser. A user logs in, then navigates to the Tags link in the navigation bar. A list of available tags should be displayed in alphabetical order. The user then clicks on the **"Pencil"** (Edit) button to edit the selected tag and is redirected to a form, which populates the tag chosen to edit. The user inputs the new tag name and clicks on the **"Update"** button, which redirects the user back to the list of tags. If a user chooses to edit a tag, then once they are in the edit form change their mind, they can click **"Return to List"**, which cancels the change request and redirects user back to the list of tags.

### [TICKET #16 Edit A Tag](https://trello.com/c/uakCoJx6/16-16-edit-a-tag)


## Type of change


# Testing Instructions

- run `git add .` then `git commit` all changes in your terminal
- `git checkout main` then `git fetch -a` to get all branches that haven't been fetched yet
- `git checkout ajb-tag-edit`
- `start ~{working-directory}/tabloidfullstack-fool-s-spring/Tabloid.sln`
- Run these queries in your SQL Object Explorer [Create DB](https://github.com/NewForce-Cohort-5/tabloidfullstack-fool-s-spring/blob/main/SQL/01_Tabloid_Create_DB.sql) and [Seed Data](https://github.com/NewForce-Cohort-5/tabloidfullstack-fool-s-spring/blob/main/SQL/02_Tabloid_Seed_Data.sql)
- In Visual Studio, run in debugger using the `Tabloid` server
- `cd ~{working-directory}/tabloidfullstack-fool-s-spring/Tabloid/client && npm start`
- Log in as 'foo@bar.com'
- Click the Tags link in the navigation bar
- You should now see a list of all tags.
- Run the SQL query to view an updated list of tags in the database: 
```sql
SELECT * FROM Tag
```
- List as it should appear in your database, minus any added or deleted tags:


| Id | Name
| --- | --- 
| 1 | C#
| 2 | JavaScript
| 3 | Funny
| 4 | Scary 
| ... | ... 

- Click on the "Pencil" icon.
- A form appears with the select tag to edit pre-populating the form field.
- Input the new tag name, then select "Update" to save the change request.
- If user chooses not to change the tag name, click the "Return to List" link to cancel the request and be redirected to the list of tags.
- Tags should be shown in alphabetical order.
- Run the SQL query again after any tag has been deleted to compare with the updated list in the browswer:
```sql
SELECT * FROM Tag
```


# Checklist:
